### ISSUE TITLE
SigningAlgorithms ignored from appSettings.json

### ISSUE THREAD

**Issue by gamblen** [DuendeSoftware/Support#753]
I am trying to configure, via appSettings.json, the SigningAlgorithms allowed.

As SigningAlgorithms is defined as
`public IEnumerable<SigningAlgorithmOptions> SigningAlgorithms { get; set; }`
with IEnumerable there is no .Add method therefore using the Options it cannot add the values I define in.  I found that information in [dotnet issue 63083](https://github.com/dotnet/runtime/issues/63083)

with this in my appSettings.json file, the values I define in SigningAlgorithms are ignored and result in only RS256
```
  "IdentityServerOptions": {
    "KeyManagement": {
      "Enabled": true,
      "SigningAlgorithms": [
        {
          "Name": "RS512",
          "UseX509Certificate": true
        },
        {
          "Name": "RS384"
        }
      ]
    }
  }
```

Am I doing something wrong?  Or is not expected to override the signing algorithms?

Thanks,

**Comment by gamblen** [comment]
Also noted that SigningAlgorithmOptions does not have a public parameterless constructor it would be ignore by the IConfiguration calls

**Comment by josephdecock** [comment]
This looks like a bug to me. While we usually configure IdentityServer in code in our samples, I think you ought to be able to supply the config from any normal configuration source. @brockallen do you agree/is that our intent?

@gamblen, are you able to work around this issue with code for now? For example:

```
        services.AddIdentityServer(options =>
        {
            options.KeyManagement.SigningAlgorithms = new[]
            {
                new SigningAlgorithmOptions("RS384"),
                new SigningAlgorithmOptions("RS512")
                {
                    UseX509Certificate = true;
                }
            }
        })
```

**Comment by brockallen** [comment]
Sure. But this might be a limitation in the .NET config system. They had issues with a collection that was already initialized. More investigation is needed @josephdecock.

**Comment by gamblen** [comment]
Yes, I actually copied the SigningAlgorithmsOptions class, removed the constructor and added it to a new MyIdentityServerOptions and used it with that section.  Then once The official bit had bind'd I replace the KeyAlgorithms with my IList version.

> This looks like a bug to me. While we usually configure IdentityServer in code in our samples, I think you ought to be able to supply the config from any normal configuration source. @brockallen do you agree/is that our intent?
> 
> @gamblen, are you able to work around this issue with code for now? For example:
> 
> ```
>         services.AddIdentityServer(options =>
>         {
>             options.KeyManagement.SigningAlgorithms = new[]
>             {
>                 new SigningAlgorithmOptions("RS384"),
>                 new SigningAlgorithmOptions("RS512")
>                 {
>                     UseX509Certificate = true;
>                 }
>             }
>         })
> ```
